File: app/src/main/java/org/watsi/uhp/custom_components/NotificationBar.java
package org.watsi.uhp.custom_components;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import org.watsi.uhp.R;

public class NotificationBar extends LinearLayout {

    // displayed message
    private String message;

    // displayed button text
    private String action;

    // internal components
    private TextView notificationMessage;
    private Button notificationBtn;

    public NotificationBar(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        initControl(context, attrs);
    }

    private void initControl(Context context, AttributeSet attrs)
    {
        LayoutInflater inflater = (LayoutInflater)
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.notification_bar, this);

        notificationMessage = (TextView)findViewById(R.id.notification_message);
        notificationBtn = (Button)findViewById(R.id.notification_btn);

        TypedArray ta = getContext().obtainStyledAttributes(attrs, R.styleable.NotificationBar);

        try {
            message = ta.getString(R.styleable.NotificationBar_message);
            action = ta.getString(R.styleable.NotificationBar_action);

            notificationMessage.setText(message);
            notificationBtn.setText(action);
        } finally {
            ta.recycle();
        }
    }

    public void setOnActionClickListener(OnClickListener l) {
        notificationBtn.setOnClickListener(l);
    }
}

File: app/src/main/java/org/watsi/uhp/views/NotificationBar.kt
package org.watsi.uhp.views

import android.content.Context
import android.support.constraint.ConstraintLayout
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import kotlinx.android.synthetic.main.notification_bar.view.notification_btn

import org.watsi.uhp.R

class NotificationBar(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {

    init {
        LayoutInflater.from(context).inflate(R.layout.notification_bar, this)

        val notificationMessage = findViewById<TextView>(R.id.notification_message)
        val notificationBtn = findViewById<Button>(R.id.notification_btn)

        val ta = getContext().obtainStyledAttributes(attrs, R.styleable.NotificationBar)

        try {
            val message = ta.getString(R.styleable.NotificationBar_message)
            val action = ta.getString(R.styleable.NotificationBar_action)

            notificationMessage.text = message
            notificationBtn.text = action
        } finally {
            ta.recycle()
        }
    }

    override fun setOnClickListener(listener: OnClickListener?) {
        notification_btn.setOnClickListener(listener)
    }

    fun setMessage(message: String, messageColor: Int? = null, backgroundColor: Int? = null) {
        val notificationMessage = findViewById<TextView>(R.id.notification_message)
        val component = findViewById<ConstraintLayout>(R.id.notification_banner)
        notificationMessage.text = message
        messageColor?.let { notificationMessage.setTextColor(it) }
        backgroundColor?.let { component.setBackgroundColor(backgroundColor)}
    }
}
